endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
+        vel_disp_multiplier: float = 1.0,
+        zero: typing.Literal["start", "mean", "median", None] = "median",
+        include_integration: bool = True,
+        include_pseudo_velocity: bool = True,
+        damp: float = 0.05,
+        init_freq: float = 1.0,
+        bins_per_octave: float = 12,
+        include_resultant: bool = False,
+        display_plots: bool = False,
+) -> pd.DataFrame:
+    """
+    Compute the following shock and vibration metrics for a given time series dataframe
+        - Peak Absolute Acceleration
+        - RMS Acceleration
+        - Peak Absolute Velocity
+        - RMS Velocity
+        - Peak Absolute Displacement
+        - RMS Displacement
+        - Peak Pseudo Velocity
+    :param df: the input dataframe with an index defining the time in seconds or datetime
+    :param tukey_percent: the portion of the time series to apply a Tukey window (a taper that forces beginning and end to 0), default is 0.1
+        - Note that the RMS metrics will only be computed on the portion of time that isn't tapered
+    :param zero: the output quantity driven to zero prior to the calculations
+        `None` does nothing
+        `"start"` forces the first datapoint to 0,
+        `"mean"` chooses ``-np.mean(output)`` & `"median"` (default) chooses
+        ``-np.median(output)``
+    :param highpass_cutoff: the cutoff frequency of a preconditioning highpass
+        filter; if None, no filter is applied. For shock events, it is recommended to set this to None (the default), but it is recommended for vibration.
+    :param vel_disp_multiplier: applies a scale to the velocity and displacement metrics
+        - 386.09 to convert from g to inches (in)
+        - 9806.65 to convert from g to millimeters (mm)
+    :param include_integration: include the calculations of velocity and displacement, default behavior
+    :param include_pseudo_velocity: include the more time consuming calculation of pseudo velocity, default behavior
+    :param damp: the damping coefficient `ζ`, related to the Q-factor by
+        `ζ = 1/(2Q)`; defaults to 0
+    :param init_freq: the initial frequency in the sequence; if `None`,
+        use the frequency corresponding to the data's duration, default is 1.0 Hz
+    :param bins_per_octave: the number of frequencies per octave
+    :param include_resultant: add a resultant (root sum of the squares) for each metric, calculated from the other input dataframe columns
+    :param display_plots: display plotly figures of the min/max envelope of acceleration, velocity, displacement and PVSS (default as False)
+    :return: a dataframe containing all the metrics, one computed per column of the input dataframe
+    """
+
+    #Remove Offset
+    if zero is not None:
+        if zero == "start":
+            df = df - df.iloc[0]
+        elif zero in ("mean", "median"):
+            zero = dict(mean=np.mean, median=np.median)[zero]
+            df = df - df.apply(zero, axis="index", raw=True)
```

Comment:
Better to keep this one level.

```suggestion
    if zero == "start":
        df -= df.iloc[0]
    elif zero == "mean":
        df -= df.mean(axis="index")
    elif zero == "median":
        df -= df.median(axis="index")
    elif zero is not None:
        raise ValueError(f'kwarg zero was {zero}, must be one of [None, "start", "mean", "median"]')
```